percona/controllers/perconapgcluster_controller.go

```diff
@@ -98,58 +99,111 @@ func (r *PGClusterReconciler) Reconcile(ctx context.Context, request reconcile.R
-		postgresCluster.Spec.DatabaseInitSQL = perconaPGCluster.Spec.DatabaseInitSQL
-		postgresCluster.Spec.Patroni = perconaPGCluster.Spec.Patroni
+		postgresCluster.Spec.Backups = cr.Spec.Backups
+		postgresCluster.Spec.DataSource = cr.Spec.DataSource
+		postgresCluster.Spec.DatabaseInitSQL = cr.Spec.DatabaseInitSQL
+		postgresCluster.Spec.Patroni = cr.Spec.Patroni
 
-		if perconaPGCluster.Spec.PMM != nil && perconaPGCluster.Spec.PMM.Enabled {
+		if cr.Spec.PMM != nil && cr.Spec.PMM.Enabled {
 			// TODO: Check if PMM secret exists
-			for i := 0; i < len(perconaPGCluster.Spec.InstanceSets); i++ {
-				set := &perconaPGCluster.Spec.InstanceSets[i]
-				set.Sidecars = append(set.Sidecars, pmm.SidecarContainer(perconaPGCluster))
+			for i := 0; i < len(cr.Spec.InstanceSets); i++ {
+				set := &cr.Spec.InstanceSets[i]
+				set.Sidecars = append(set.Sidecars, pmm.SidecarContainer(cr))
 			}
 		}
 
-		postgresCluster.Spec.InstanceSets = perconaPGCluster.Spec.InstanceSets.ToCrunchy()
-		postgresCluster.Spec.Users = perconaPGCluster.Spec.Users
-		postgresCluster.Spec.Proxy = perconaPGCluster.Spec.Proxy.ToCrunchy()
+		postgresCluster.Spec.InstanceSets = cr.Spec.InstanceSets.ToCrunchy()
+		postgresCluster.Spec.Users = cr.Spec.Users
+		postgresCluster.Spec.Proxy = cr.Spec.Proxy.ToCrunchy()
 
 		return nil
 	})
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "update/create PostgresCluster")
+	}
 
 	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(postgresCluster), postgresCluster); err != nil {
 		return reconcile.Result{}, errors.Wrap(err, "get PostgresCluster")
 	}
 
-	perconaPGCluster.Status = v2beta1.PerconaPGClusterStatus{
+	host, err := r.getHost(ctx, cr)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "get app host")
+	}
+
+	cr.Status = v2beta1.PerconaPGClusterStatus{
+		State:                 r.updateState(cr, &postgresCluster.Status),
+		Host:                  host,
 		PostgresClusterStatus: postgresCluster.Status,
 	}
 
-	if err := r.Client.Status().Update(ctx, perconaPGCluster); err != nil {
+	if err := r.Client.Status().Update(ctx, cr); err != nil {
 		log.Error(err, "failed to update status")
 	}
 
 	return reconcile.Result{}, err
 }
 
+func (r *PGClusterReconciler) getHost(ctx context.Context, cr *v2beta1.PerconaPGCluster) (string, error) {
+	svcName := cr.Name + "-pgbouncer"
+
+	if cr.Spec.Proxy.PGBouncer.ServiceExpose == nil || cr.Spec.Proxy.PGBouncer.ServiceExpose.Type != string(corev1.ServiceTypeLoadBalancer) {
+		return svcName + "." + cr.Namespace, nil
+	}
+
+	svc := &corev1.Service{}
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: cr.Namespace, Name: svcName}, svc)
+	if err != nil {
+		return "", errors.Wrapf(err, "get %s service", svcName)
+	}
+
+	var host string
+	for _, i := range svc.Status.LoadBalancer.Ingress {
+		host = i.IP
+		if len(i.Hostname) > 0 {
+			host = i.Hostname
+		}
+	}
+
+	return host, nil
+}
+
+func (r *PGClusterReconciler) updateState(cr *v2beta1.PerconaPGCluster,
```

Comment:
`updateState` doesn't make too much sense since it doesn't update anything. `getState` maybe?